rsocket-examples/src/main/java/io/rsocket/examples/transport/tcp/lease/RateLimitingWithLeaseExample.java

```diff
@@ -0,0 +1,211 @@
+import io.rsocket.core.RSocketServer;
+import io.rsocket.lease.Lease;
+import io.rsocket.lease.LeaseStats;
+import io.rsocket.lease.Leases;
+import io.rsocket.lease.MissingLeaseException;
+import io.rsocket.transport.netty.client.TcpClientTransport;
+import io.rsocket.transport.netty.server.CloseableChannel;
+import io.rsocket.transport.netty.server.TcpServerTransport;
+import io.rsocket.util.ByteBufPayload;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.ReplayProcessor;
+import reactor.util.retry.Retry;
+
+public class RateLimitingWithLeaseExample {
+  private static final String SERVER_TAG = "server";
+  private static final String CLIENT_TAG = "client";
+
+  public static void main(String[] args) {
+    // Queue for incoming messages represented as Flux
+    // Imagine that every fireAndForget that is pushed is processed by a worker
+
+    int queueCapacity = 50;
+    BlockingQueue<String> messagesQueue = new ArrayBlockingQueue<>(queueCapacity);
+
+    // emulating a worker that process data from the queue
+    Thread workerThread =
+        new Thread(
+            () -> {
+              try {
+                while (!Thread.currentThread().isInterrupted()) {
+                  String message = messagesQueue.poll(Long.MAX_VALUE, TimeUnit.DAYS);
+                  System.out.println("Process message {" + message + "}");
+                  Thread.sleep(500); // emulating processing
+                }
+              } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+              }
+            });
+
+    workerThread.start();
+
+    CloseableChannel server =
+        RSocketServer.create(
+                (setup, sendingSocket) ->
+                    Mono.just(
+                        new RSocket() {
+                          @Override
+                          public Mono<Void> fireAndForget(Payload payload) {
+                            // add element. if overflows errors and terminates execution
+                            // specifically to show that lease can limit rate of fnf requests in
+                            // that example
+                            try {
+                              messagesQueue.add(payload.getDataUtf8());
+                            } catch (IllegalStateException e) {
+                              e.printStackTrace();
+                              sendingSocket.dispose();
+                              workerThread.interrupt();
+                              throw e;
+                            }
+                            payload.release();
+                            return Mono.empty();
```

Comment:
What about using `offer()` instead? Either way "payload.release()" should be in a `finally`.
```suggestion
                            try {
                              if (!messagesQueue.offer(payload.getDataUtf8())) {
                                // TODO: log error message
                                sendingSocket.dispose();
                                workerThread.interrupt();
                              }
                            } finally {
                              payload.release();
                            }
                            return Mono.empty();
```